### PR TITLE
Adds contextControl prop to <TopBar /> and <Navigation />

### PR DESCRIPTION
**Pull request by kvendrik**
### WHY are these changes introduced?

As part of the context switching updates ([details](https://github.com/Shopify/web/issues/9665)) this PR proposes a `contextControl` prop thats is added to the `<TopBar />` and `<Navigation />` components and accepts a component that is ideally used to provide a way for users to switch between admin contexts.

### WHAT is this pull request doing?
Adds the `contextControl` prop to `<TopBar />` and `<Navigation />`.

### How to 🎩
<details>
<summary>1. Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import * as React from 'react';
import {TopBar, AppProvider, Frame, Navigation, Button} from '../src';

interface State {}

export default class Playground extends React.Component<{}, State> {
  render() {
    const topBar = (
      <TopBar
        contextControl={
          <TopBar.Menu
            action={[]}
            activatorContent="Click here to switch from one context to another"
          />
        }
      />
    );

    const nav = (
      <Navigation
        location="/"
        contextControl={
          <Button>Click here to switch from one context to another</Button>
        }
      />
    );

    return (
      <AppProvider
        theme={{
          logo: {
            width: 124,
            topBarSource:
              'https://cdn.shopify.com/s/files/1/0446/6937/files/jaded-pixel-logo-color.svg?6215648040070010999',
          },
        }}
      >
        <Frame topBar={topBar} navigation={nav} showMobileNavigation />
      </AppProvider>
    );
  }
}
```

</details>

2. Run `yarn dev` and navigate to `http://localhost:6006/iframe.html?selectedKind=Playground&selectedStory=Playground`.
3. Make sure the `<TopBar.Menu />` shows up in the far left corner of the top bar.
4. Remove the `contextControl` prop from the top bar.
5. Make sure the logo shows up and looks/works as you would expect.
6. Make sure the `Button` that was passed into `Navigation` doesn't show up on the page.
6. Resize the screen to a mobile size.
7. Make sure the `Button` that was passed into `Navigation` shows up in the side nav.

### CODE REVIEW

File: src/components/TopBar/tests/TopBar.test.tsx [hunk 1]
```diff
@@ -226,6 +226,39 @@ describe('<TopBar />', () => {
       expect(topBar.find(UnstyledLink).prop('style')).toEqual({width: '104px'});
     });
   });
+
+  describe('contextControl', () => {
+    const mockContextControl = (
+      <TopBar.Menu
+        actions={[]}
+        activatorContent="Switch stores"
```
Comment (tmlayton) [inline src/components/TopBar/tests/TopBar.test.tsx]:
```suggestion
        activatorContent="Switch contexts"
```
🕵️‍♀️

File: src/components/TopBar/TopBar.tsx [hunk 2]
```diff
@@ -83,20 +86,25 @@ export class TopBar extends React.PureComponent<ComposedProps, State> {
...
-        <Image
-          source={logo.topBarSource || ''}
-          alt={logo.accessibilityLabel || ''}
-          className={styles.Logo}
-          style={{width}}
-        />
-      </UnstyledLink>
-    ) : null;
+    const contextMarkup =
+      !contextControl && logo ? (
```
Comment (tmlayton) [inline src/components/TopBar/TopBar.tsx]: We will want to use if-else flow control here:

if context control -> render context control
else if logo -> render logo
else -> null

Comment (kvendrik) [reply src/components/TopBar/TopBar.tsx]: Makes sense that makes it a bit clearer and resolves the no `contextControl` nor `logo` case.

**Comment by kvendrik** [comment]
> We are missing a test case for top bar without context control and without logo. Currently that code path with still render and empty context control div.

Yes good catch! Will fix that!

**Comment by danrosenthal** [comment]
During my :tophat: I noticed this was able to be gotten into a weird state if the initial render was on desktop and you resize it to mobile, where the `Backdrop` was visible but not the `Navigation`. When initial render is on mobile, the `Navigation` renders as expected. This was pretty weird, but I don't see how it could be related to these changes.

**Pull request merged**